x-pack/plugins/watcher/server/routes/api/watch/action/register_acknowledge_route.ts

```diff
@@ -56,14 +54,8 @@ export function registerAcknowledgeRoute({
           body: { watchStatus: watchStatus.downstreamJson },
         });
       } catch (e) {
-        // Case: Error from Elasticsearch JS client
-        if (isEsError(e)) {
-          const body = e.statusCode === 404 ? `Watch with id = ${watchId} not found` : e;
-          return response.customError({ statusCode: e.statusCode, body });
-        }
-
-        // Case: default
-        throw e;
+        // TODO: Figure out if this covers us sufficiently given that previous logic returned a body with "Watch with id = ${watchId} not found" previously
+        return handleEsError({ error: e, response });
```

Comment:
Did you look into how to test this, either in this handler or in the other route handlers that contain this `TODO`?

Reply:
Same as above, will re-add

Reply:
Yeah, I just checked this out and it looks like the ES response we get looks like:

<img width="343" alt="Screenshot 2021-04-29 at 15 56 29" src="https://user-images.githubusercontent.com/8155004/116565906-90d97380-a906-11eb-976f-c8166ae0d50a.png">

Which is not very user friendly. So I reintroduced the renaming logic so that the response for 404s is now:

<img width="350" alt="Screenshot 2021-04-29 at 15 59 45" src="https://user-images.githubusercontent.com/8155004/116565983-a189e980-a906-11eb-90c3-91ae5437e08f.png">

Reply:
Nice!